olp-cpp-sdk-core/src/thread/ThreadPoolTaskScheduler.cpp

```diff
@@ -82,9 +82,7 @@ class ThreadPoolTaskScheduler::QueueImpl {
 
   bool Pull(ElementType& element) { return sync_queue_.Pull(element); }
 
-  void Push(ElementType&& element) {
-    sync_queue_.Push(std::forward<ElementType>(element));
-  }
+  void Push(ElementType&& element) { sync_queue_.Push(std::move(element)); }
```

Comment:
std::forward ?

Reply:
For perfect forwarding we need a templated method (here we have a templated class).
E.g. `template<typename U> void Push(...){...}`

Reply:
But `ThreadPoolTaskScheduler::QueueImpl` is not a templated class also, so `std::forward` is just a `std::move` here.